### PR TITLE
Split Client and Add Express Server\n\nSplit client code into subdire…

### PR DESCRIPTION
**Pull request by Zarex**
…ctory and added express server to root level.  Updated package.json and added build options for client and server separately

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +12,8 @@
 # production
 /build
 /dist
+/dist2
```
Comment (Misterimpulse) [inline .gitignore:15]: Why dist 2?

File: babel.config.js [hunk 2]
```diff
@@ -5,18 +5,8 @@ module.exports = (api) => {
...
-    '@babel/transform-runtime',
-    '@babel/plugin-transform-object-assign',
-    '@babel/plugin-transform-modules-commonjs',
-    '@babel/plugin-proposal-object-rest-spread',
-    '@babel/plugin-proposal-class-properties',
-    '@babel/plugin-transform-react-jsx-source',
-    '@babel/plugin-transform-regenerator',
-    ['babel-plugin-webpack-alias', { config: path.resolve(__dirname, './relativePath.config.js') }],
   ];
+  const plugins = [];
```
Comment (Misterimpulse) [inline babel.config.js:9]: Don't think we need this empty array?

File: client/src/index.html [hunk 3]
```diff
@@ -6,12 +6,11 @@
    <meta name="viewport" content="width=device-width, user-scalable=0, maximum-scale=1, minimum-scale=1" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <link href="https://fonts.googleapis.com/css?family=Lato&display=swap" rel="stylesheet">
-   <title>React App</title>
+   <title>React CRApp</title>
```
Comment (Misterimpulse) [inline client/src/index.html:9]: -_-

File: client/src/logo.svg [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+        <circle cx="420.9" cy="296.5" r="45.7"/>
+        <path d="M520.5 78.1z"/>
+    </g>
+</svg>
```
Comment (Misterimpulse) [inline client/src/logo.svg:7]: Do we need this file still?

File: client/webpack.client.config.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+const path = require('path');
+// TODO: prod vs dev
+// TODO: Add back in relative path
```
Comment (Misterimpulse) [inline client/webpack.client.config.js:3]: Delete todos, don't need them anymore

File: package.json [hunk 6]
```diff
@@ -7,22 +7,22 @@
     "url": "git+https://github.com/Kansas-City-Mutual-Aid/kc-covid.git"
   },
   "scripts": {
-    "build": "npm run clean:build && cp src/index.html dist/index.html && npm run pack",
+    "build": "npm run clean:build && cp src/html/index.html dist/index.html && npm run pack",
```
Comment (Misterimpulse) [inline package.json:10]: I think we want this to run clean:build, build:server, build:client

File: package.json [hunk 7]
```diff
@@ -7,22 +7,22 @@
...
+    "build:server": "rm -rf dist && webpack --mode development --config webpack.config.js",
+    "build:client": "cd ./client && npm run build",
     "build-win": "powershell -Command \"Copy-Item 'src/index.html' 'dist/index.html'\" && npm run pack-win",
-    "clean": "rimraf package-lock.json && rimraf ./node_modules",
-    "clean:build": "if [ -d './dist' ]; then rimraf ./dist; fi && mkdir dist",
```
Comment (Misterimpulse) [inline package.json:13]: Hey put this back

File: package.json [hunk 8]
```diff
@@ -7,22 +7,22 @@
     "url": "git+https://github.com/Kansas-City-Mutual-Aid/kc-covid.git"
   },
   "scripts": {
-    "build": "npm run clean:build && cp src/index.html dist/index.html && npm run pack",
+    "build": "npm run clean:build && cp src/html/index.html dist/index.html && npm run pack",
+    "build:server": "rm -rf dist && webpack --mode development --config webpack.config.js",
```
Comment (Misterimpulse) [inline package.json:11]: We want this to run clean:build and then webpack in prod mode

File: package.json (hunk 6 above)
Comment (Misterimpulse) [inline package.json:10]: We will want a second build "build:dev" to build everything in development mode. This should basically be a copy of build but in development mode

File: package.json [hunk 9]
```diff
@@ -7,22 +7,22 @@
...
     "pack-win": "webpack --config webpack.config.js -p",
-    "postinstall": "npm run build",
-    "start": "webpack-dev-server --mode development",
-    "test": "npm run test:lint && npm run test:jest",
-    "lint:scss": "stylelint src/**/*.scss",
-    "test:lint": "eslint --ext .js,.jsx, . && npm run lint:scss",
-    "test:jest": "jest --config jestconfig.json"
+    "start": "node ./dist/server.js",
+    "start:dev": "cd ./client && npm run start",
+    "postinstall": "find ./* -maxdepth 1 -mindepth 1 -name package.json -execdir npm install \\;"
```
Comment (Misterimpulse) [inline package.json:25]: Wat is this doing?

File: package.json [hunk 10]
```diff
@@ -7,22 +7,22 @@
...
-    "clean": "rimraf package-lock.json && rimraf ./node_modules",
-    "clean:build": "if [ -d './dist' ]; then rimraf ./dist; fi && mkdir dist",
+    "clean": "npm run clean:client && npm run clean:server",
+    "clean:client": "cd ./client && npm run clean",
+    "clean:server": "rimraf package-lock.json && rimraf ./node_modules",
     "clean:install": "npm run clean && npm install",
     "compile": "npm run compile:clean && npm run compile:build",
     "compile:clean": "if [ -d './src' ]; then rimraf ./lib; fi",
     "compile:build": "babel src --out-dir lib --copy-files",
     "pack": "NODE_ENV=production webpack --config webpack.config.js -p",
```
Comment (Misterimpulse) [inline package.json:21]: Delete

File: client/package.json [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+  "scripts": {
+    "build": "npm run clean:build && cp src/index.html dist/index.html && npm run pack",
+    "build-win": "powershell -Command \"Copy-Item 'src/index.html' 'dist/index.html'\" && npm run pack-win",
+    "clean": "rimraf package-lock.json && rimraf ./node_modules",
+    "clean:build": "if [ -d './dist' ]; then rimraf ./dist; fi && mkdir dist",
+    "clean:install": "npm run clean && npm install",
+    "compile": "npm run compile:clean && npm run compile:build",
+    "compile:clean": "if [ -d './src' ]; then rimraf ./lib; fi",
+    "compile:build": "babel src --out-dir lib --copy-files",
+    "pack": "NODE_ENV=production webpack --config webpack.client.config.js -p",
```
Comment (Misterimpulse) [inline client/package.json:18]: Delete

File: client/package.json [hunk 12]
```diff
@@ -0,0 +1,103 @@
+{
+  "name": "kc-covid-client",
+  "version": "0.1.0",
+  "private": true,
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Kansas-City-Mutual-Aid/kc-covid.git"
+  },
+  "scripts": {
+    "build": "npm run clean:build && cp src/index.html dist/index.html && npm run pack",
```
Comment (Misterimpulse) [inline client/package.json:10]: instead of "run pack" do "run build:production"

Comment (Misterimpulse) [inline client/package.json:10]: Add a build:dev command to be called from the top level

File: client/package.json [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+    "build-win": "powershell -Command \"Copy-Item 'src/index.html' 'dist/index.html'\" && npm run pack-win",
+    "clean": "rimraf package-lock.json && rimraf ./node_modules",
```
Comment (Misterimpulse) [inline client/package.json:12]: Don't need the package-lock part anymore

File: client/package.json [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+    "clean:build": "if [ -d './dist' ]; then rimraf ./dist; fi && mkdir dist",
+    "clean:install": "npm run clean && npm install",
+    "compile": "npm run compile:clean && npm run compile:build",
+    "compile:clean": "if [ -d './src' ]; then rimraf ./lib; fi",
+    "compile:build": "babel src --out-dir lib --copy-files",
```
Comment (Misterimpulse) [inline client/package.json:17]: Don't need these

File: package.json [hunk 15]
```diff
@@ -7,22 +7,22 @@
...
     "build-win": "powershell -Command \"Copy-Item 'src/index.html' 'dist/index.html'\" && npm run pack-win",
-    "clean": "rimraf package-lock.json && rimraf ./node_modules",
-    "clean:build": "if [ -d './dist' ]; then rimraf ./dist; fi && mkdir dist",
+    "clean": "npm run clean:client && npm run clean:server",
+    "clean:client": "cd ./client && npm run clean",
+    "clean:server": "rimraf package-lock.json && rimraf ./node_modules",
     "clean:install": "npm run clean && npm install",
     "compile": "npm run compile:clean && npm run compile:build",
     "compile:clean": "if [ -d './src' ]; then rimraf ./lib; fi",
     "compile:build": "babel src --out-dir lib --copy-files",
```
Comment (Misterimpulse) [inline package.json:20]: We can remove these

File: .gitignore (hunk 1 above)
Comment (Zarex) [reply .gitignore:15]: removed

File: package.json [hunk 16]
```diff
@@ -37,62 +37,31 @@
...
+    "eslint": "^6.8.0",
+    "eslint-loader": "^4.0.0",
     "eslint-plugin-react": "^7.19.0",
     "file-loader": "^6.0.0",
-    "html-loader": "^1.0.0",
-    "html-webpack-plugin": "^3.2.0",
-    "jest": "25.1.0",
-    "jest-matchers": "^20.0.3",
-    "jest-snapshot": "^25.1.0",
-    "mocha": "^7.1.1",
```
Comment (Misterimpulse) [inline package.json:84]: Still need these

File: babel.config.js (hunk 2 above)
Comment (Zarex) [reply babel.config.js:9]: removed

File: package.json [hunk 17]
```diff
@@ -37,62 +37,31 @@
...
+    "@babel/preset-env": "^7.9.0",
+    "babel-eslint": "^10.1.0",
+    "babel-loader": "^8.1.0",
     "css-loader": "^3.4.2",
-    "enzyme": "^3.11.0",
-    "enzyme-adapter-react-16": "^1.15.2",
-    "enzyme-react-intl": "^2.0.6",
-    "enzyme-to-json": "^3.4.4",
-    "eslint": "^5.8.0",
-    "eslint-config-airbnb": "^18.1.0",
```
Comment (Misterimpulse) [inline package.json:72]: Need this

File: package.json [hunk 18]
```diff
@@ -37,62 +37,31 @@
...
-    "eslint-import-resolver-alias": "^1.1.2",
-    "eslint-plugin-import": "^2.20.1",
-    "eslint-plugin-jest": "^23.8.2",
```
Comment (Misterimpulse) [inline package.json:75]: Need

File: client/src/index.html (hunk 3 above)
Comment (Zarex) [reply client/src/index.html:9]: I apologize only for leaving it in there after testing.

File: package.json [hunk 19]
```diff
@@ -37,62 +37,31 @@
...
-    "eslint-plugin-jsx-a11y": "^6.2.3",
```
Comment (Misterimpulse) [inline package.json:76]: Need

File: client/src/logo.svg (hunk 4 above)
Comment (Zarex) [reply client/src/logo.svg:7]: gone

File: client/webpack.client.config.js (hunk 5 above)
Comment (Zarex) [reply client/webpack.client.config.js:3]: done

File: src/server/server-dev.js [hunk 20]
```diff
@@ -0,0 +1,54 @@
+// import path from 'path'
```
Comment (Misterimpulse) [inline src/server/server-dev.js:1]: Do we need this file? If not remove

File: webpack.config.js [hunk 21]
```diff
@@ -1,32 +1,60 @@
...
-};
+module.exports = (env, argv) => {
+  const SERVER_PATH = (argv.mode === 'production') ?
+    './src/server/server-prod.js' :
+    './src/server/server-prod.js'//'./src/server/server-dev.js'
+    
+return ({
+    entry: {
+      server: SERVER_PATH,
+    },
```
Comment (Misterimpulse) [inline webpack.config.js:14]: Don't need this logic :)

File: webpack.config.js [hunk 22]
```diff
@@ -1,32 +1,60 @@
...
+      ]
+    },
+    plugins: [
+      new HtmlWebPackPlugin({
+        template: "./src/html/index.html",
+        filename: "./index.html",
+        excludeChunks: [ 'server' ]
+      }),
+      //new webpack.HotModuleReplacementPlugin(),
+      //new webpack.NoEmitOnErrorsPlugin()
```
Comment (Misterimpulse) [inline webpack.config.js:57]: Remove

File: webpack.config.js [hunk 23]
```diff
@@ -1,32 +1,60 @@
...
+          }
+        },
+        {
+          // Loads the javacript into html template provided.
+          // Entry point is set below in HtmlWebPackPlugin in Plugins 
+          test: /\.html$/,
+          use: [
+            {
+              loader: "html-loader",
+              //options: { minimize: true }
```
Comment (Misterimpulse) [inline webpack.config.js:44]: Remove

File: package.json (hunk 6 above)
Comment (Zarex) [reply package.json:10]: yeah, good. Okay

File: package.json (hunk 7 above)
Comment (Zarex) [reply package.json:13]: done

File: client/package.json [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+    "pack": "NODE_ENV=production webpack --config webpack.client.config.js -p",
+    "pack-win": "webpack --config webpack.client.config.js -p",
+    "postinstall": "npm run build",
+    "start": "webpack-dev-server --mode development",
```
Comment (Misterimpulse) [inline client/package.json:21]: By default we want to start in production

File: package.json (hunk 8 above)
Comment (Zarex) [reply package.json:11]: done

File: package.json (hunk 9 above)
Comment (Zarex) [reply package.json:25]: confusing everybody who reads it.  Removed for now

File: webpack.config.js [hunk 25]
```diff
@@ -1,32 +1,60 @@
-const path = require('path');
+const path = require('path')
```
Comment (Misterimpulse) [inline webpack.config.js:1]: I'll leave most of this file alone since I assume you'll change it with your next story

File: package.json (hunk 10 above)
Comment (Zarex) [reply package.json:21]: done

File: client/package.json (hunk 11 above)
Comment (Zarex) [reply client/package.json:18]: done

File: client/package.json (hunk 12 above)
Comment (Zarex) [reply client/package.json:10]: done

Comment (Zarex) [reply client/package.json:10]: done

File: package.json (hunk 9 above)
Comment (Misterimpulse) [reply package.json:25]: We will want this to run install on the client and run build on the server

File: client/package.json (hunk 13 above)
Comment (Zarex) [reply client/package.json:12]: BEGONE

File: client/package.json (hunk 14 above)
Comment (Zarex) [reply client/package.json:17]: bye

File: package.json (hunk 15 above)
Comment (Zarex) [reply package.json:20]: buh
bye

File: package.json (hunk 17 above)
Comment (Zarex) [reply package.json:72]: done

File: package.json (hunk 18 above)
Comment (Zarex) [reply package.json:75]: done

File: package.json (hunk 19 above)
Comment (Zarex) [reply package.json:76]: done

File: src/server/server-dev.js (hunk 20 above)
Comment (Zarex) [reply src/server/server-dev.js:1]: removed

File: webpack.config.js (hunk 21 above)
Comment (Zarex) [reply webpack.config.js:14]: yeet

File: webpack.config.js (hunk 22 above)
Comment (Zarex) [reply webpack.config.js:57]: yeeterino

File: webpack.config.js (hunk 23 above)
Comment (Zarex) [reply webpack.config.js:44]: blamo

File: webpack.config.js (hunk 25 above)
Comment (Zarex) [reply webpack.config.js:1]: ty bb

File: client/package.json (hunk 24 above)
Comment (Zarex) [reply client/package.json:21]: done

**Pull request merged**